skimage/restoration/rolling_ball.py

```diff
@@ -0,0 +1,184 @@
+        same number of dimensions as ``image``. Kernel is filled with the
+        intensity of the kernel at that position.
+    nansafe: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+    num_threads: int, optional
+        The maximum number of threads to use. If ``None`` use the OpenMP
+        default value; typically equal to the maximum number of virtual cores.
+        Note: This is an upper limit to the number of threads. The exact number
+        is determined by the system's OpenMP library.
+
+    Returns
+    -------
+    background : ndarray
+        The estimated background of the image.
+
+    Notes
+    -----
+    For the pixel that has its background intensity estimated (without loss
+    of generality at ``midpoint``) the rolling ball method centers ``kernel``
+    under it and raises the kernel until its surface touches the image umbra
+    at some ``pos=(y,x)``. The background intensity is then estimated
+    using the image intensity at that position (``image[pos]``) plus the
+    difference of ``kernel[center] - kernel[pos]``.
+
+    This algorithm assumes that dark pixels correspond to the background. If
+    you have a bright background, invert the image before passing it to the
+    function, e.g., using `utils.invert`. See the gallery example for details.
+
+    This algorithm is sensitive to noise (in particular salt-and-pepper
+    noise). If this is a problem in your image, you can apply mild
+    gaussian smoothing before passing the image to this function.
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.restoration import rolling_ball
+    >>> image = data.coins()
+    >>> background = rolling_ball(data.coins())
+    >>> filtered_image = image - background
+
+
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.restoration import rolling_ball, ellipsoid_kernel
+    >>> image = data.coins()
+    >>> kernel = ellipsoid_kernel((101, 101), 75)
+    >>> background = rolling_ball(data.coins())
+    >>> filtered_image = image - background   
+    """
+
+    image = np.asarray(image)
+    img = image.astype(np.float_)
+
+    if num_threads is None:
+        num_threads = 0
+
+    if kernel is None:
+        kernel = ball_kernel(radius, image.ndim)
+
+    kernel_shape = np.asarray(kernel.shape)
+    kernel_center = (kernel_shape // 2)
+    center_intensity = kernel[tuple(kernel_center)]
+
+    intensity_difference = center_intensity - kernel
+    intensity_difference[kernel == np.Inf] = np.Inf
+    intensity_difference = intensity_difference.astype(img.dtype)
+    intensity_difference = intensity_difference.ravel()
```

Comment:
```suggestion
    intensity_difference = intensity_difference.reshape(-1)
```
As advised in the note in [`numpy.ravel`'s doc](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html)